packages/node-app/src/utils/validations.utils.ts

```diff
@@ -54,9 +64,12 @@ const validateUnit = (participant, giveaway) => {
   const requiredUnit = giveaway.requirements.unit;
 
   // unit required and valid or unit not required
-  if ((requiredUnit && requiredUnit === participantUnit) || !requiredUnit)
-    return ParticipantState.CONFIRMED;
-  return ParticipantState.REJECTED;
+  if (requiredUnit) {
+    const found = participantUnit.some((unit) => unit === requiredUnit);
+    return found ? ParticipantState.CONFIRMED : ParticipantState.REJECTED;
```

Comment:
No need to loop through all user units
```suggestion
    return participantUnits.includes(requiredUnit)
      ? ParticipantState.CONFIRMED
      : ParticipantState.REJECTED;
```